packages/office-ui-fabric-react/src/components/Button/PrimaryButton/PrimaryButton.tsx

```diff
@@ -1,18 +1,35 @@
+import * as React from 'react';
+import { BaseComponent, customizable, nullRender } from '@uifabric/utilities';
 import { BaseButton } from '../BaseButton';
+import { IButtonProps, IButtonClassNames } from '../Button.Props';
+
 import styles = require('./PrimaryButton.scss');
 
-export class PrimaryButton extends BaseButton {
-  protected classNames = {
-    base: 'ms-Button',
-    variant: 'ms-Button--primary',
-    icon: styles.icon,
-    menuIcon: styles.icon,
-    isDisabled: styles.isDisabled,
-    isEnabled: styles.isEnabled,
-    label: styles.label,
-    root: styles.root
-  };
+const CLASS_NAMES: IButtonClassNames = {
+  base: 'ms-Button',
+  variant: 'ms-Button--primary',
+  icon: styles.icon,
+  menuIcon: styles.icon,
+  isDisabled: styles.isDisabled,
+  isEnabled: styles.isEnabled,
+  label: styles.label,
+  root: styles.root
+};
 
-  protected onRenderDescription() { return null; }
+@customizable('PrimaryButton')
+export class PrimaryButton extends BaseComponent<IButtonProps, {}> {
+  /**
+   * Set this BaseComponent._resolveComponentRef to false, bypassing resolution of componentRef.
+   */
+  protected _shouldUpdateComponentRef = false;
 
+  public render() {
+    return (
+      <BaseButton
+        classNames={ CLASS_NAMES }
```

Comment:
little nit, but this one one CLASS_NAMES and icon is IconButtonClassNames. Can we stick to 1 convention?